src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -758,9 +759,11 @@ void MacroAssembler::resolve_global_jobject(Register value, Register tmp1, Regis
 }
 
 void MacroAssembler::stop(const char* msg) {
-  BLOCK_COMMENT(msg);
+  // Skip AOT caching C strings in scratch buffer.
+  const char* str = (code_section()->scratch_emit()) ? msg : AOTCodeCache::add_C_string(msg);
+  BLOCK_COMMENT(str);
   illegal_instruction(Assembler::csr::time);
-  emit_int64((uintptr_t)msg);
+  emit_int64((uintptr_t)str);
```

Comment:
I noticed that our aarch64 counterpart puts the address of msg into `r0` in https://github.com/openjdk/jdk/pull/24740.
```
  // load msg into r0 so we can access it from the signal handler
  // ExternalAddress enables saving and restoring via the code cache
  lea(c_rarg0, ExternalAddress((address) str));
```

And fetches the address from `r0` in `PosixSignals::pd_hotspot_signal_handler`:
```
        // A pointer to the message will have been placed in `r0`
        const char *detail_msg = (const char *)(uc->uc_mcontext.regs[0]);
```

I am not sure but I guess this is needed for the correct working of aot? Maybe we should do similar things.